projects/project-player-library/src/lib/pages/certificate-page/certificate-page.component.ts

```diff
@@ -107,31 +118,45 @@ export class CertificatePageComponent extends BackNavigationHandlerComponent {
       const scale = dpi / 96;
       canvas.width = width * scale;
       canvas.height = height * scale;
+    if (ctx) {
       ctx?.scale(scale, scale);
       ctx?.drawImage(img, 0, 0, width, height);
       if (type === 'pdf') {
-        const pdf = new jsPDF({
-          orientation: width > height ? 'landscape' : 'portrait',
-          unit: 'pt',
-          format: [width, height],
-        });
-        const imgData = canvas.toDataURL('image/png');
-        pdf.addImage(imgData, 'PNG', 0, 0, width, height);
-        pdf.save(`${this.projectDetails.title}.pdf`);
-        URL.revokeObjectURL(url);
+        this.downloadPdf();
       } else {
+        this.downloadPng(canvas);
+      }
+    }
+  }
+
+  downloadPdf() {
+    fetch(this.projectDetails.certificate.pdfUrl)
```

Comment:
Here without converting it into blob, can we use the same url from api response and achieve the same functionality?

Reply:
creating a temporary file to rename the file without using that name of the file is not changing.

Reply:
ok